cmd/acr/purge.go

```diff
@@ -614,8 +617,65 @@ func dryRunPurge(ctx context.Context, acrClient api.AcrCLIClientInterface, login
 	return deletedTagsCount, deletedManifestsCount, nil
 }
 
+// getDoNotDeleteListFromMultiArchManifestWithTag checks if a manifest has tags and its media type support multiarch manifests.
+// If so, the dependent manifests are marked not to be deleted
+func getDoNotDeleteListFromMultiArchManifestWithTag(ctx context.Context, manifest acr.ManifestAttributesBase, doNotDelete set.Set[string], acrClient api.AcrCLIClientInterface, repoName string) error {
+	switch *manifest.MediaType {
+	case mediaTypeDockerManifestList, v1.MediaTypeImageIndex:
+		var manifestBytes []byte
+		manifestBytes, err := acrClient.GetManifest(ctx, repoName, *manifest.Digest)
+		if err != nil {
+			return err
+		}
+		// this struct defines a customized struct for manifests
+		// which is used to parse the content of a multiarch manifest
+		mam := struct {
+			Manifests []v1.Descriptor `json:"manifests"`
+		}{}
+
+		if err = json.Unmarshal(manifestBytes, &mam); err != nil {
+			return err
+		}
+		for _, dependentManifest := range mam.Manifests {
+			doNotDelete.Add(dependentManifest.Digest.String())
+		}
+	default:
+		doNotDelete.Add(*manifest.Digest)
```

Comment:
Do we need this? It looks like the original code did not add the manifest itself to `dotNotDelete`.